core/src/main/scala/chisel3/Reg.scala

```diff
@@ -127,7 +127,7 @@ object RegNext {
   * val x = Wire(UInt())
   * val y = Wire(UInt(8.W))
   * val r1 = RegInit(x) // width will be inferred
-  * val r2 = RegInit(y) // width is set to 8
+  * val r2 = RegInit(y) // width will be inferred
```

Comment:
Isn't it set to 8 eventually? What's the mechanism?

Reply:
inference in FIRRTL

Reply:
Can we add it as a comment? It may not be obvious that it is inferred, especially if the example code doesn't assign anything else to r2 that could change its width.

Reply:
Here's the [scastie demo](https://scastie.scala-lang.org/b159bXm4R2yCkchFAzjFRA). I am not sure of the mechanism, but it confirms, I believe, the correctness of Jack's change

Reply:
Read the whole ScalaDoc, that’s sort of the point of this whole example: https://www.chisel-lang.org/api/3.4.1/chisel3/RegInit$.html

It’s just that that particular comment is wrong and contradicts the correct info above.